packages/core/src/codewhispererChat/tools/executeBash.ts

```diff
@@ -46,22 +153,60 @@ export class ExecuteBash {
         }
     }
 
-    public requiresAcceptance(): boolean {
+    public requiresAcceptance(): CommandValidation {
         try {
-            const args = ExecuteBash.parseCommand(this.command)
+            const args = split(this.command)
             if (!args || args.length === 0) {
-                return true
+                return { requiresAcceptance: true, isHighRisk: false }
+            }
+
+            // Split commands by pipe and process each segment
+            let currentCmd: string[] = []
+            const allCommands: string[][] = []
+
+            for (const arg of args) {
+                if (arg === '|') {
+                    if (currentCmd.length > 0) {
+                        allCommands.push(currentCmd)
+                    }
+                    currentCmd = []
+                } else if (arg.includes('|')) {
+                    return { requiresAcceptance: true, isHighRisk: false }
+                } else {
+                    currentCmd.push(arg)
+                }
             }
 
-            if (args.some((arg) => dangerousPatterns.some((pattern) => arg.includes(pattern)))) {
-                return true
+            if (currentCmd.length > 0) {
+                allCommands.push(currentCmd)
             }
 
-            const command = args[0]
-            return !readOnlyCommands.includes(command)
+            for (const cmdArgs of allCommands) {
+                if (cmdArgs.length === 0) {
+                    return { requiresAcceptance: true, isHighRisk: false }
+                }
+
+                if (cmdArgs.some((arg) => Array.from(dangerousPatterns).some((pattern) => arg.includes(pattern)))) {
+                    return { requiresAcceptance: true, isHighRisk: false }
```

Comment:
isn't it possible for high risk command to contain dangerous pattern? This logic would make that not high risk anymore?

Reply:
Updated to make dangerousPatterns high risk